x/staging/posts/keeper/common_functions.go

```diff
@@ -32,25 +33,25 @@ func (k Keeper) IteratePosts(ctx sdk.Context, fn func(index int64, post types.Po
 func (k Keeper) ValidatePost(ctx sdk.Context, post types.Post) error {
 	params := k.GetParams(ctx)
 	maxMsgLen := params.MaxPostMessageLength.Int64()
-	maxOpFieldNum := params.MaxAdditionalAttributesFieldsNumber.Int64()
-	maxOpFieldValLen := params.MaxAdditionalAttributesFieldValueLength.Int64()
+	maxAddAttrsNum := params.MaxAdditionalAttributesFieldsNumber.Int64()
+	maxAddAttrValLen := params.MaxAdditionalAttributesFieldValueLength.Int64()
 
 	if int64(len(post.Message)) > maxMsgLen {
-		return sdkerrors.Wrap(sdkerrors.ErrInvalidRequest,
+		return sdkerrors.Wrap(types.ErrMessageLengthExceeded,
```

Comment:
I think this and the below errors can be all something more generic like `ErrInvalidPost` instead. This would allow use to have fewer error types and use `sdkerrors.Wrap` to specify what's wrong, allowing the clients to handle them better